template_yarte/src/main.rs

```diff
@@ -1,29 +1,28 @@
 use std::collections::HashMap;
 
 use actix_web::{
-    error::ErrorInternalServerError, get, middleware::Logger, web, App, Error,
-    HttpResponse, HttpServer,
+    get, middleware::Logger, web, App, Error, HttpResponse, HttpServer, ResponseError,
 };
-use yarte::TemplateMin;
+use derive_more::Display;
+use yarte::ywrite_min;
 
-#[derive(TemplateMin)]
-#[template(path = "index")]
-struct IndexTemplate {
-    query: web::Query<HashMap<String, String>>,
-}
+#[derive(Debug, Display)]
+struct MyErr(pub &'static str);
+
+impl ResponseError for MyErr {}
 
 #[get("/")]
 async fn index(
     query: web::Query<HashMap<String, String>>,
 ) -> Result<HttpResponse, Error> {
-    IndexTemplate { query }
-        .call()
-        .map(|body| {
-            HttpResponse::Ok()
-                .content_type("text/html; charset=utf-8")
-                .body(body)
-        })
-        .map_err(|_| ErrorInternalServerError("Some error message"))
+    // TODO: check this capacity
+    let mut body = web::BytesMut::with_capacity(4096);
+    // `ywrite_min` is work in progress check your templates before put in production or use `ywrite_html`
+    ywrite_min!(body, "{{> index }}");
```

Comment:
so, this still ends up using `query` correctly?

Reply:
yes, in short what it does is copy code inside a `{}` block. It can have any input and any propagation `?`.

In the template `index.hbs` uses `query.get(...)` any `query` that has `get` method is valid. It's tested :)

Reply:
nice